Remove lexicographical sorting of storage keys in torch.save

mikaylagawarecki opened the pull request:
Currently the order lexicographical (i.e. 0, 10, 11, ...19, 2, ....) instead of 0, 1, 2, 3, 4, 5 (the order that storage metadata is actually pickled in), since PyTorch will never be used with Python < 3.7 we can be assured that the keys will be read in the order of insertion (numerically sorted)

This makes it such that the order storages are written in are the same as the pickling/unpickling order so we can calculate their offsets with less random reads


Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #143880
* __->__ #143879

mikaylagawarecki committed:
Remove lexicographical sorting of storage keys in torch.save

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Remove lexicographical sorting of storage keys in torch.save"




[ghstack-poisoned]

mikaylagawarecki commented:
@mikaylagawarecki has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D67673025).

mikaylagawarecki commented:
@mikaylagawarecki has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D67673025).

mikaylagawarecki commented:
@mikaylagawarecki has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D67673025).

mikaylagawarecki committed:
Update on "Remove lexicographical sorting of storage keys in torch.save"

Currently the order lexicographical (i.e. 0, 10, 11, ...19, 2, ....) instead of 0, 1, 2, 3, 4, 5 (the order that storage metadata is actually pickled in), this makes it such that the order storages are written in are the same as the pickling/unpickling order so we can calculate their offsets with less random reads



Differential Revision: [D67673025](https://our.internmc.facebook.com/intern/diff/D67673025)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Remove lexicographical sorting of storage keys in torch.save"

Currently the order lexicographical (i.e. 0, 10, 11, ...19, 2, ....) instead of 0, 1, 2, 3, 4, 5 (the order that storage metadata is actually pickled in), this makes it such that the order storages are written in are the same as the pickling/unpickling order so we can calculate their offsets with less random reads



Differential Revision: [D67673025](https://our.internmc.facebook.com/intern/diff/D67673025)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Remove lexicographical sorting of storage keys in torch.save"

Currently the order lexicographical (i.e. 0, 10, 11, ...19, 2, ....) instead of 0, 1, 2, 3, 4, 5 (the order that storage metadata is actually pickled in), this makes it such that the order storages are written in are the same as the pickling/unpickling order so we can calculate their offsets with less random reads



Differential Revision: [D67673025](https://our.internmc.facebook.com/intern/diff/D67673025)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Remove lexicographical sorting of storage keys in torch.save"

Currently the order lexicographical (i.e. 0, 10, 11, ...19, 2, ....) instead of 0, 1, 2, 3, 4, 5 (the order that storage metadata is actually pickled in), this makes it such that the order storages are written in are the same as the pickling/unpickling order so we can calculate their offsets with less random reads



Differential Revision: [D67673025](https://our.internmc.facebook.com/intern/diff/D67673025)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Remove lexicographical sorting of storage keys in torch.save"

Currently the order lexicographical (i.e. 0, 10, 11, ...19, 2, ....) instead of 0, 1, 2, 3, 4, 5 (the order that storage metadata is actually pickled in), this makes it such that the order storages are written in are the same as the pickling/unpickling order so we can calculate their offsets with less random reads



Differential Revision: [D67673025](https://our.internmc.facebook.com/intern/diff/D67673025)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Remove lexicographical sorting of storage keys in torch.save"

Currently the order lexicographical (i.e. 0, 10, 11, ...19, 2, ....) instead of 0, 1, 2, 3, 4, 5 (the order that storage metadata is actually pickled in), this makes it such that the order storages are written in are the same as the pickling/unpickling order so we can calculate their offsets with less random reads



Differential Revision: [D67673025](https://our.internmc.facebook.com/intern/diff/D67673025)

[ghstack-poisoned]

albanD reviewed (approved):
Sounds good!

mikaylagawarecki committed:
Update on "Remove lexicographical sorting of storage keys in torch.save"

Currently the order lexicographical (i.e. 0, 10, 11, ...19, 2, ....) instead of 0, 1, 2, 3, 4, 5 (the order that storage metadata is actually pickled in), since PyTorch will never be used with Python < 3.7 we can be assured that the keys will be read in the order of insertion (numerically sorted)

This makes it such that the order storages are written in are the same as the pickling/unpickling order so we can calculate their offsets with less random reads



Differential Revision: [D67673025](https://our.internmc.facebook.com/intern/diff/D67673025)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Remove lexicographical sorting of storage keys in torch.save"

Currently the order lexicographical (i.e. 0, 10, 11, ...19, 2, ....) instead of 0, 1, 2, 3, 4, 5 (the order that storage metadata is actually pickled in), since PyTorch will never be used with Python < 3.7 we can be assured that the keys will be read in the order of insertion (numerically sorted)

This makes it such that the order storages are written in are the same as the pickling/unpickling order so we can calculate their offsets with less random reads



Differential Revision: [D67673025](https://our.internmc.facebook.com/intern/diff/D67673025)

[ghstack-poisoned]

ZainRizvi commented:
@pytorchbot revert -m ghfirst -c "Sorry but this is breaking internally. See D68746524 for details"

ZainRizvi commented:
@pytorchbot revert -c ghfirst -m "Sorry but this is breaking internally. See D68746524 for details"

mikaylagawarecki committed:
Update on "Remove lexicographical sorting of storage keys in torch.save"

Currently the order lexicographical (i.e. 0, 10, 11, ...19, 2, ....) instead of 0, 1, 2, 3, 4, 5 (the order that storage metadata is actually pickled in), since PyTorch will never be used with Python < 3.7 we can be assured that the keys will be read in the order of insertion (numerically sorted)

This makes it such that the order storages are written in are the same as the pickling/unpickling order so we can calculate their offsets with less random reads



Differential Revision: [D67673025](https://our.internmc.facebook.com/intern/diff/D67673025)

[ghstack-poisoned]

mikaylagawarecki commented:
@mikaylagawarecki has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D68901268).

albanD reviewed (approved):
Thanks!

mikaylagawarecki commented:
@pytorchbot merge -f "landed internally"

mikaylagawarecki commented:
@pytorchbot merge -f "landed internally"

mikaylagawarecki commented:
@pytorchbot merge -i "landed internally"

mikaylagawarecki commented:
@pytorchbot merge -i

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented:
@pytorchbot merge -f

mikaylagawarecki commented:
@pytorchbot merge -f "landed internally"
